client/packages/programs/src/JsonForms/common/components/Number.tsx

```diff
@@ -14,17 +20,42 @@ import {
 import { Box } from '@mui/system';
 import { FormLabel } from '@mui/material';
 
-export const numberTester = rankWith(3, schemaTypeIs('number'));
+export const numberTester = rankWith(
+  5,
+  or(schemaTypeIs('number'), uiTypeIs('Number'))
+);
+
+type NumberOptions = {
+  /*
+  Options only required if the saved data should be converted to a string
+  representation of the number
+  */
+  output?: 'string';
+  /*
+  The output string will be padded with leading 0's up to the number of significant figures specified
+  */
+  sigFigs?: number;
```

Comment:
In `packages/programs/src/JsonForms/components/IdGenerator.tsx`

there is already a padString option. Think its cleaner to use it there and limit what the Number control can do.

For example:
```json
                    {
                        "type": "Field",
                        "field": "birthOrder",
                        "mutations": [{ "firstNChars": 2 }, { "padString": "00" }]
                      },
```

(might need a number -> string conversion in IdGenerator)

Reply:
Hmm, in which case, should we just add a number-> string conversion into the ID generator instead?

I think if we're going to allow the Number control to save a string, then it needs to be format-able, otherwise what's the point of it? But now that I think about it, maybe we should just leave the number control as a number and convert it in the ID generator?

That would mean changing the schema to make "Birth Order" a numeric value instead of a string, but that actually makes more sense, doesn't it?

Reply:
Yes sounds good!